hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java

```diff
@@ -496,12 +504,72 @@ SqlResult execute(ExplainStatementPlan plan) {
 
     SqlResult execute(SelectPlan plan, QueryId queryId, List<Object> arguments, long timeout) {
         List<Object> args = prepareArguments(plan.getParameterMetadata(), arguments);
+        InternalSerializationService serializationService = Util.getSerializationService(hazelcastInstance);
+        ExpressionEvalContext evalContext = new ExpressionEvalContextImpl(
+                args,
+                serializationService,
+                Util.getNodeEngine(hazelcastInstance));
+
         JobConfig jobConfig = new JobConfig()
                 .setArgument(SQL_ARGUMENTS_KEY_NAME, args)
                 .setArgument(KEY_SQL_QUERY_TEXT, plan.getQuery())
                 .setArgument(KEY_SQL_UNBOUNDED, plan.isStreaming())
                 .setTimeoutMillis(timeout);
 
+        final Set<Integer> partitions = new HashSet<>();
+        boolean allVariantsValid = true;
+        for (final String mapName : plan.getPartitionStrategyCandidates().keySet()) {
+            var perMapCandidates = plan.getPartitionStrategyCandidates().get(mapName);
+            final var container = getMapContainer(hazelcastInstance.getMap(mapName));
+            final PartitioningStrategy<?> strategy = container.getPartitioningStrategy();
+
+            // We only support Default and Attribute strategies, even if one of the maps uses non-Default/Attribute
+            // strategy, we should abort the process and clear list of already populated partitions so that partition
+            // pruning doesn't get activated at all for this query.
+            if (!(strategy instanceof DefaultPartitioningStrategy)
```

Comment:
why do we check this (again?) so late? If partitioning strategy for the IMap is not supported, information about it is availabe during the optimization process. Does it protect against race condition in case strategy for IMap is changed? BTW, if so, partitioning strategy and attributes should be a part of `PartitionedMapPlanObjectKey` to invalidate plans in such case.

Reply:
@ivanthescientist

Reply:
@Fly-Style you can fix it as well

Reply:
@ivanthescientist @Fly-Style what are we doing with that?

Reply:
ah, it is fixed, ok.